Library/Homebrew/utils/github.rb

```diff
@@ -495,9 +494,7 @@ def fetch_artifact(user, repo, pr, dir,
     # preferred over system `curl` and `tar` as this leverages the Homebrew
     # cache to avoid repeated downloads of (possibly large) bottles.
     FileUtils.chdir dir do
-      curl_args[:cache] = Pathname.new(dir)
-      curl_args[:secrets] = [token]
-      downloader = strategy.new(artifact_url, "artifact", pr, **curl_args)
+      downloader = strategy.new(artifact_url, "artifact", pr, curl_args: curl_args, secrets: [token])
```

Comment:
Thoughts on moving `GitHubArtifactDownloadStrategy` to here and not passing in `strategy`? Is there a use case for a customising the strategy further?

Reply:
Given the way `curl_args` are handled now, I don't think it would even work properly on the default `CurlDownloadStrategy`.

Reply:
@Bo98 I want to avoid `GitHubArtifactDownloadStrategy` being in the global (formula) namespace. I'm open to any solution for that.

Reply:
Yeah, fair point, let's just move it into utils/github.rb

Reply:
@MikeMcQuaid That makes sense, but what's the specific case where `GitHubArtifactDownloadStrategy` might get dragged into formulae etc when it shouldn't? Is making the class private etc sufficient?

Reply:
By "private" I mean stuff like `private_constant :GitHubArtifactDownloadStrategy`.